azkaban-common/src/main/java/azkaban/project/Dependency.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+import azkaban.Constants;
+import azkaban.utils.Props;
+import com.google.common.base.Preconditions;
+
+/**
+ * Dependency is an immutable class which holds
+ * all the data and properties of a dependency.
+ */
+public class Dependency {
```

Comment:
Do you think it's better to name it like DataDependency or TriggerDependency? Because we already have similar naming for some other concepts like job dependency or flow dependency.

Reply:
it depends on where to put those classes. Previously I thought they should be in a package called trigger. but now if it needs to be in the same package as AzkabanFlow, then i will update the name.